WooCommerce/Classes/ViewRelated/Orders/OrdersMasterViewController.swift

```diff
@@ -0,0 +1,218 @@
+
+import UIKit
+import struct Yosemite.OrderStatus
+import struct Yosemite.Note
+
+/// The main Orders view controller that is shown when the Orders tab is accessed.
+///
+/// TODO This should contain the tabs "Processing" and "All Orders".
+///
+final class OrdersMasterViewController: UIViewController {
+
+    private lazy var analytics = ServiceLocator.analytics
+
+    private lazy var viewModel = OrdersMasterViewModel(statusFilterChanged: { [weak self] (status: OrderStatus?) in
+        self?.statusFilterChanged(status: status)
+    })
+
+    /// The view controller that shows the list of Orders.
+    ///
+    private var ordersViewController: OrdersViewController?
+
+    init() {
+        super.init(nibName: Self.nibName, bundle: nil)
+
+        title = NSLocalizedString("Orders", comment: "The title of the Orders tab.")
+
+        tabBarItem.title = title
+        tabBarItem.image = .pagesImage
+        tabBarItem.accessibilityIdentifier = "tab-bar-orders-item"
```

Comment:
I have some concerns about that pattern. For someone new to the project, I ran into the pattern quite a lot. Some things on the top of my mind:

- `configure*` is ambiguous. It breaks the common convention that function names should clearly state what they do. 
- Since `configure*` is ambiguous and its purpose is not clear, it eventually encourages me to add more stuff to it later. This will just exacerbate the issue. 
- More methods mean decreased comprehension and increased time to grok the class. Every additional method adds more jumping around when analyzing what the class does. 
- The methods add noise to the class' accessible properties and methods. Generally, `configure*` methods are not intended to be reusable. But since they're methods, they're always there. Eventually, I will make the mistake of _reusing_ them.

There's still more to this and I actually couldn't find a good alternative. I'm still trying to wrap my head around it. 

But yeah, I do understand the value of making it consistent so I'll change it for now until we can have some more discussions about it.

Reply:
Added `configureTabBarItem()` in c88f7176136150c2395c530eee4067834c24ebb7.

Reply:
I agree that these setup functions aren't reusable since they should only be called once, I think breaking them to functions does declutter `viewDidLoad`/init quite a bit and specify the intention for each setup in the function name at the cost of having to jump to the function for details. Personally I don't have strong opinions on this, I'm fine with either way (the other way: having sections of code, perhaps with a comment on the top of each section). I suggested as a nit (non-blocking) mostly out of consistency. Would you like to write a post with the points above? I'm curious to hear what people think 😄 cc @pmusolino

Reply:
> I think breaking them to functions does declutter viewDidLoad/init quite a bit and specify the intention for each setup in the function name at the cost of having to jump to the function for details

Yup. To be clear, I'm not against splitting them into functions either. I just wish my concerns are acknowledged too. 🙂 

> Would you like to write a post with the points above?

It's in my long **p2-posts-to-write backlog**. 😄I wanted to find some solutions beforehand though. But yeah.